### PR TITLE
New nick replacement mode

### PR DESCRIPTION
**Pull request by ForNeVeR**
Guys have given me an advice me to improve the nick replacer in this way. I've also written a test.

### CODE REVIEW

File: src/main/scala/ru/org/codingteam/horta/protocol/jabber/MucMessageHandler.scala [hunk 1]
```diff
@@ -129,10 +129,10 @@ class MucMessageHandler(locale: LocaleDefinition,
     } else {
       var message = text
       for (nick <- participants.keys.map(Protocol.nickByJid)) {
-        if (nick != recipient && nick.length > 0) {
+        if (nick != recipient && nick.length > 2) {
```
Comment (rexim) [inline src/main/scala/ru/org/codingteam/horta/protocol/jabber/MucMessageHandler.scala]: So the people with nick length less or equal 2 are guilty by definition?

Comment (ForNeVeR) [inline src/main/scala/ru/org/codingteam/horta/protocol/jabber/MucMessageHandler.scala]: Yes they are.

**Comment by rexim** [comment]
Is something wrong with the tests?

**Comment by ForNeVeR** [comment]
That's very strange, gimme some time to investigate.

**Comment by rexim** [comment]
Lol, what's the difference between merged and unmerged branch? :)

File: src/main/scala/ru/org/codingteam/horta/protocol/jabber/MucMessageHandler.scala [hunk 2]
```diff
@@ -164,5 +164,23 @@ class MucMessageHandler(locale: LocaleDefinition,
       core ! CoreParticipantLeft(Clock.now, roomJID, participantJID, reason, self)
     }
   }
+}
+
+object MucMessageHandler {
 
+  private val vowels = "(?i)[aeiouаоэиуыеёюя]".r // TODO: Use some phonetic definition instead of hardcode
```
Comment (rexim) [inline src/main/scala/ru/org/codingteam/horta/protocol/jabber/MucMessageHandler.scala]: I would even recommend to create an issue for that TODO

File: src/main/scala/ru/org/codingteam/horta/protocol/jabber/MucMessageHandler.scala [hunk 3]
```diff
@@ -164,5 +164,23 @@ class MucMessageHandler(locale: LocaleDefinition,
...
+  private val dash = "-"
```
Comment (rexim) [inline src/main/scala/ru/org/codingteam/horta/protocol/jabber/MucMessageHandler.scala]: Why didn't you call the variable `separator` or `delimiter`? I mean what's the point to call the variable by its value. I'd rather call it by its purpose.

File: src/main/scala/ru/org/codingteam/horta/protocol/jabber/MucMessageHandler.scala [hunk 4]
```diff
@@ -164,5 +164,23 @@ class MucMessageHandler(locale: LocaleDefinition,
...
+
+  def getNickReplacement(nick: String): String = {
+    if (nick.length <= 1) {
+      nick
+    } else {
+      val replacement = vowels.replaceFirstIn(nick, dash)
```
Comment (rexim) [inline src/main/scala/ru/org/codingteam/horta/protocol/jabber/MucMessageHandler.scala]: So, now the definition of being guilty is not having any vowels. Ok.

File: src/main/scala/ru/org/codingteam/horta/protocol/jabber/MucMessageHandler.scala (hunk 3 above)
Comment (ForNeVeR) [inline src/main/scala/ru/org/codingteam/horta/protocol/jabber/MucMessageHandler.scala]: What a wundurful ideah! :3

File: src/main/scala/ru/org/codingteam/horta/protocol/jabber/MucMessageHandler.scala (hunk 2 above)
Comment (ForNeVeR) [inline src/main/scala/ru/org/codingteam/horta/protocol/jabber/MucMessageHandler.scala]: I will create the issue before merging that to `develop`.

**Comment by rexim** [comment]
Ok, I approve this shit!

**Comment by ForNeVeR** [comment]
Thank you for so useful review.

**Pull request merged**